src/data/majors/cs.ts

```diff
@@ -186,3 +172,86 @@ export const csAdvisors: AdvisorGroup = {
   ],
   source: 'https://www.cs.cornell.edu/undergrad/ustaff/contact-academic-advisor',
 };
+
+export const csMigrations: RequirementMigration[] = [
+  {
+    entryYear: 2023, // CS students entering Fall 2024 or later have new core requirements (listed above)
+    type: 'Modify',
+    fieldName: 'Computer Science Core',
+    newValue: {
+      name: 'Computer Science Core',
+      description: 'CS 2800 or CS 2802, CS 3110, CS 3410 or CS 3420, CS 4410, and CS 4820',
+      source: 'https://www.cs.cornell.edu/undergrad/csmajor',
+      checker: includesWithSubRequirements(
+        ['CS 2800', 'CS 2802'],
+        ['CS 3110'],
+        ['CS 3410', 'CS 3420'],
+        ['CS 4820'],
+        ['CS 4410', 'CS 4414']
+      ),
+      fulfilledBy: 'courses',
+      perSlotMinCount: [1, 1, 1, 1, 1],
+      slotNames: [
+        'CS 2800 or CS 2802',
+        'CS 3110',
+        'CS 3410 or CS 3420',
+        'CS 4820',
+        'CS 4410 or CS 4414',
+      ],
+    },
+  },
+  {
+    entryYear: 2023, // CS students entering Fall 2024 or later have new CS elective requirements (listed above)
+    type: 'Modify',
+    fieldName: 'CS Electives',
+    newValue: {
+      name: 'CS Electives',
+      description:
+        'Three 4000+ CS electives each at 3 credits. CS 4090, CS 4998, and CS 4999 are NOT allowed.',
+      source:
+        'http://www.cs.cornell.edu/undergrad/rulesandproceduresengineering/choosingyourelectives',
+      checker: [
+        (course: Course): boolean => {
+          if (
+            courseMatchesCodeOptions(course, ['CS 4090', 'CS 4998', 'CS 4999', 'CS 4410', 'CS 4820'])
+          ) {
+            return false;
+          }
+          return (
+            ifCodeMatch(course.subject, 'CS') &&
+            !(
+              ifCodeMatch(course.catalogNbr, '1***') ||
+              ifCodeMatch(course.catalogNbr, '2***') ||
+              ifCodeMatch(course.catalogNbr, '3***')
+            )
+          );
+        },
+      ],
+      fulfilledBy: 'courses',
+      perSlotMinCount: [3],
+      slotNames: ['Course'],
+    },
+  },
+  {
+    entryYear: 2023, // CS students entering Fall 2024 or later no longer have a Probability requirement
+    type: 'Add',
```

Comment:
Great work with adding these requirement migrations, especially as you don't have access to our documentation! Splendid code here, and good that you added the comments.